web/client-api/src/main/java/io/deephaven/web/client/api/JsPartitionedTable.java

```diff
@@ -81,16 +82,15 @@ public Promise<JsPartitionedTable> refetch() {
 
             return w.getExportedObjects()[0].fetch();
         }).then(result -> connection.newState((c, state, metadata) -> {
-            JsTable keyTable = (JsTable) result;
-            DropColumnsRequest drop = new DropColumnsRequest();
-            drop.setColumnNamesList(new String[] {descriptor.getConstituentColumnName()});
-            drop.setSourceId(keyTable.state().getHandle().makeTableReference());
-            drop.setResultId(state.getHandle().makeTicket());
-            connection.tableServiceClient().dropColumns(drop, metadata, (fail, success) -> {
-                keyTable.close();
+            baseTable = (JsTable) result;
+            SelectOrUpdateRequest view = new SelectOrUpdateRequest();
+            view.setSourceId(baseTable.state().getHandle().makeTableReference());
+            view.setResultId(state.getHandle().makeTicket());
+            view.setColumnSpecsList(descriptor.getKeyColumnNamesList());
+            connection.tableServiceClient().view(view, metadata, (fail, success) -> {
                 c.apply(fail, success);
             });
-        }, "drop constituent column")
+        }, "get underlying table")
```

Comment:
You already had the underlying table at line 85. This operation is creating a view that's just the keys table.
```suggestion
        }, "get keys table")
```